src/commands.cpp

```diff
@@ -8319,33 +8360,43 @@ bool CommandBackupRemove::procresult(Result r)
 {
     client->app->backupremove_result(r.errorOrOK(), mBackupId);
 
-    if (r.succeeded())  // remove the corresponding backup name from the user's attribute
-    {
-        User *ownUser = client->finduser(client->me);
-        attr_t attrtype = ATTR_BACKUP_NAMES;
+
+    // Upon removal of backup successfully --> remove the backup name silently for the user's attribute
+    if (r.succeeded())
+    {        
         std::string key {Base64Str<MegaClient::BACKUPHANDLE>(mBackupId)};
-        const std::string *oldValue = ownUser->getattr(attrtype);
-        if (!oldValue)  // attr doesn't exist -> create it
+        attr_t attrType = ATTR_BACKUP_NAMES;
+
+        User *ownUser = client->finduser(client->me);
+        const std::string *oldValue = ownUser->getattr(attrType);
+
+        if (oldValue && !ownUser->isattrvalid(attrType)) // not fetched yet or outdated
         {
-            LOG_warn << "Backup was removed, but there was no name for it. Backup id: " << key;
+            LOG_warn << "Cannot immediately remvoe backup name for backup id: " << key << ". Fetching...";
+            client->getua(ownUser, attrType, 0);
+            client->mSendingBackupName = true;
         }
-        else if (!ownUser->isattrvalid(attrtype))
+
+        if (client->mSendingBackupName)
         {
-            LOG_err << "Failed to remove backup name for backup id: " << key;
+            // accumulate this update for the future, in order to avoid race conditions
+            // they will be sent upon `upv` completion for the update in progress
+            client->mPendingBackupNames[key] = Base64::btoa("");
         }
-        else
+        else if (oldValue)  // in the event of non-existing attribute, nothing to update
         {
-            string_map attrMap;
-            attrMap[key] = "";
+            // send backup name for this backup directly
+            std::unique_ptr<TLVstore> tlv { TLVstore::containerToTLVrecords(oldValue, &client->key) };
 
-            std::unique_ptr<TLVstore> tlv;
-            tlv.reset(TLVstore::containerToTLVrecords(oldValue, &client->key));
-
-            if (User::mergeUserAttribute(attrtype, attrMap, *tlv.get()))
+            string_map attrMap = move(client->mPendingBackupNames);
```

Comment:
same as before. I also wonder if it isn't worth it to encapsulate all the code that gets the old attribute, create the tlv and merges the pending changes (addition/removal), and calls putua with the new data.

Reply:
I think it doesn't worth. Each case is slightly different.